table/column.go

```diff
@@ -265,6 +265,8 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo, r
 		if innCasted, exit, innErr := handleZeroDatetime(ctx, col, casted, val.GetString(), types.ErrWrongValue.Equal(err)); exit {
 			return innCasted, innErr
 		}
+	} else if (sc.InInsertStmt || sc.InUpdateStmt) && err != nil && col.FieldType.Tp == mysql.TypeYear {
```

Comment:
Maybe using`types.ErrInvalidYear.Equal(err)` instead of `err != nil` will be better.

Reply:
I have modified it, thanks. @mmyj